src/interpreter/type_cast.rs

```diff
@@ -0,0 +1,170 @@
+                }
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+            if let Some(number) = value.downcast_ref::<i64>() {
+                if *number < 0 {
+                    return Err(PactConversionErr::NegativeInteger);
+                }
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+            if let Some(number) = value.downcast_ref::<i128>() {
+                if *number < 0 {
+                    return Err(PactConversionErr::NegativeInteger);
+                }
+                if *number > std::u64::MAX as i128 {
+                    return Err(PactConversionErr::Overflow);
+                }
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+        }
+
+        // Unsigned integer type casting into PactType
+        if value.is::<u8>()
+            || value.is::<u16>()
+            || value.is::<u32>()
+            || value.is::<u64>()
+            || value.is::<u128>()
+        {
+            if let Some(number) = value.downcast_ref::<u8>() {
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+            if let Some(number) = value.downcast_ref::<u16>() {
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+            if let Some(number) = value.downcast_ref::<u32>() {
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+            if let Some(number) = value.downcast_ref::<u64>() {
+                return Ok(PactType::Numeric(Numeric(*number)));
+            }
+            if let Some(number) = value.downcast_ref::<u128>() {
+                if *number > std::u64::MAX as u128 {
+                    return Err(PactConversionErr::Overflow);
+                }
+                if let Ok(n) = u64::try_from(*number) {
+                    return Ok(PactType::Numeric(Numeric(n)));
+                }
+            }
+        }
+
+        // String-like type casting into PactType
+        if value.is::<&str>() || value.is::<String>() {
+            if let Some(string) = value.downcast_ref::<&str>() {
+                return Ok(PactType::StringLike(StringLike(&*string.as_bytes())));
+            }
+            if let Some(string) = value.downcast_ref::<String>() {
+                return Ok(PactType::StringLike(StringLike(string.as_bytes())));
+            }
+        }
+
+        // Unhandled Type
+        Err(PactConversionErr::UnknownType)
+    }
+}
+
+#[cfg(test)]
```

Comment:
If a test like this can work, then there is still value in the `AnyTryInto` for associated types
```
trait Foo {
    type Number;
}

struct Bar;

impl Foo for Bar {
    type Number = u32;
}

let n = <Bar as Foo>::Number = 10u32;

assert_eq!(
    PactType::any_try_into(n)
    Ok(PactType::Numeric(Numeric(10)))
)
```